streamduck-core/src/config.rs

```diff
@@ -320,17 +322,28 @@ impl Config {
         let path = self.device_config_path();
         fs::create_dir_all(&path).await.ok();
 
-        for (serial, device) in devices.iter() {
-            let device= device.clone();
+        for (_serial, device) in devices.iter() {
+            let device = device.clone();
             self.update_collection(&device).await;
-            let mut file_path = path.clone();
-            file_path.push(format!("{}.json", serial));
-            fs::write(file_path, serde_json::to_string(device.read().await.deref()).unwrap()).await?;
+            self.write_to_filesystem(device).await?
         }
 
         Ok(())
     }
 
+    pub async fn write_to_filesystem(&self, device: UniqueDeviceConfig) -> Result<(), ConfigError> {
+        let mut path = self.device_config_path();
+        let device_conf = device.read().await;
+        path.push(format!("{}.json", device_conf.serial));
+        fs::write(path, serde_json::to_string(device_conf.deref()).unwrap()).await?;
+
+        drop(device_conf);
+        let mut device_conf = device.write().await;
+        device_conf.mark_clean();
```

Comment:
Could just reuse the lock instead of requesting new
```suggestion
        let mut path = self.device_config_path();
        let device_conf = device.write().await;
        path.push(format!("{}.json", device_conf.serial));
        fs::write(path, serde_json::to_string(device_conf.deref()).unwrap()).await?;
        device_conf.mark_clean();
```